nixos/modules/services/misc/etcd.nix

```diff
@@ -159,7 +159,8 @@ in {
     systemd.services.etcd = {
       description = "etcd key-value store";
       wantedBy = [ "multi-user.target" ];
-      after = [ "network.target" ];
+      after = [ "firewall.service" "network-online.target" ];
```

Comment:
I think that for firewall.service needs an if, not everybody may be running it.

Also, I think a note on changelog/release notes for the next version is good (if you need a reference https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_and_managing_networking/systemd-network-targets-and-services_configuring-and-managing-networking), not sure how this change impacts the use of etcd on nixpkgs (for example, Kubernetes).

Reply:
The optional for the firewall makes sense, will update.